src/core/Akka.Remote/RemoteWatcher.cs

```diff
@@ -489,8 +489,17 @@ private void ReceiveHeartbeatRsp(int uid)
 
             if (WatcheeByNodes.ContainsKey(from) && !Unreachable.Contains(from))
             {
-                if (!_addressUids.ContainsKey(from) || _addressUids[from] != uid)
+                int addressUid;
+                if (_addressUids.TryGetValue(from, out addressUid))
```

Comment:
That could cause a bug in certain circumstances, I think. 
I'm not sure, but I'd like to keep it like that for now.